src/server/shell/decoration/renderer.cpp

```diff
@@ -464,36 +449,43 @@ auto msd::Renderer::Text::Impl::utf8_to_utf32(std::string const& text) -> std::u
     return utf32_text;
 }
 
+namespace
+{
+auto init_button_drawing_functions(msd::ButtonTheme button_icons)
+{
+    using namespace msd;
+
+    std::map<ButtonFunction, std::pair<msd::Icon const, msd::Icon::DrawingFunction const>> buttons;
+
+    for(auto const& [button_function, icon]: button_icons)
+    {
+        // A little lookup table, more legible than a switch case?
+        static auto const drawing_functions = std::map<ButtonFunction, msd::Icon::DrawingFunction>{
+            {ButtonFunction::Close, render_close_icon},
+            {ButtonFunction::Maximize, render_maximize_icon},
+            {ButtonFunction::Minimize, render_minimize_icon},
+        };
+
+        buttons.emplace(button_function, std::make_pair(icon, drawing_functions.at(button_function)));
+    }
+
+    return buttons;
+}
```

Comment:
I see no reason to have the icon, button function, and icon drawing function split. A list of tuples should do?